src/python/Services/WMStats/WMStatsReader.py

```diff
@@ -4,14 +4,63 @@
 Description: General API for reading data from WMStats
 """
 
+import os
 import logging
+from logging import Logger
+from collections import defaultdict
+
+from Utilities.WebTools import getResponse
+from Utilities.ConfigurationHandler import ConfigurationHandler
+
+from typing import Optional
+
 
 class WMStatsReader(object):
     """
     _WMStats_
     General API for reading data from WMStats
     """
 
-    def __init__(self, url, logger=None, **contact):
-        pass
+    def __init__(self, logger: Optional[Logger] = None, **contact):
+        try:
+            configurationHandler = ConfigurationHandler()
+            self.reqmgrUrl = os.getenv("REQMGR_URL", configurationHandler.get("reqmgr_url"))
+            self.wmstatsEndpoint = {
+                "agentInfo": "/couchdb/wmstats/_design/WMStats/_view/agentInfo?stale=update_after"
```

Comment:
Hi, Alan! I'd like to ask you a question, is there any other endpoint that can replace this call against `couchdb`?

@amaltaro

Reply:
Hi, Amanda. Nope, there is no alternative for this one, so we need to keep calling couchdb directly.